### PR TITLE
Advanced voltage settings

### PR DESCRIPTION
**Pull request by playday3008**
Please try and fill out this template where possible, not all fields are required and can be removed.

* **Please check if the PR fulfills these requirements**
- [x] The changes have been tested locally (Only on TS100)
- [x] There are no breaking changes

* **What kind of change does this PR introduce?**
Feature

* **What is the new behavior (if this is a feature change)?**
Added customizable cut-off voltage per cell
3S - 3.0V-3.7V
4/5/6S - 2.4V-3.7V

But I have some questions
How this works, what's mean 9 and 33 numbers here
https://github.com/playday3008/IronOS/blob/9e77b84a14d188d6651b9fc0a1e40d2eabe1c477/source/Core/Threads/GUIThread.cpp#L156-L171

### CODE REVIEW

**Comment by Ralim** [comment]
33 =3.3V as using X10 readings.

The 9 is used as we divide the graphic for how charged the battery is up into chunks for how much is drawn, and so 9 is the magic number for drawing the battery graphic

I'll need to review changes tomorrow (on phone at the moment)

**Comment by playday3008** [comment]
> 33 =3.3V as using X10 readings.

This will work if I change `33` to `systemSettings.minVoltageCells`

File: Translations/make_translation.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ def getConstants():
     consants.append(("SymbolWatts", "W"))
     consants.append(("SymbolVolts", "V"))
     consants.append(("SymbolDC", "DC"))
+    consants.append(("SymbolNA", "N/A"))
```
Comment (Ralim) [inline Translations/make_translation.py:103]: Do we expect this to be a constant across all languages worldwide? Might make more sense to move to language file..?

**Review by Ralim — COMMENTED** [review]
Looks good overall, tiny comment but not sure how we want to go on the NA symbol.

I feel it may be better to suppress the option when in DC mode for the input voltage? (Is if selected just move the current menu on to skip it).

Have you tested this on hardware ?

**Comment by playday3008** [comment]
> I feel it may be better to suppress the option when in DC mode for the input voltage? (Is if selected just move the current menu on to skip it).

1 sub-item in the menu, it seems strange to me

**Comment by playday3008** [comment]
> Have you tested this on hardware ?

Only on TS100 with laboratory power supply

**Comment by Ralim** [comment]
> 
> 
> > I feel it may be better to suppress the option when in DC mode for the input voltage? (Is if selected just move the current menu on to skip it).
> 
> 1 sub-item in the menu, it seems strange to me

Yeah thats valid; that would _also_ be weird. Agree with leaving as-is then.

**Comment by Ralim** [comment]
In the menu this is called "maximum" voltage; I _believe_ this was meant to be "minimum". Otherwise seems to test well for me here.

**Review by Ralim — APPROVED** [review]

**Comment by Ralim** [comment]
🎉 Glorious! ❤️

**Comment by playday3008** [comment]
Well, Ill go to do something great (or bad)
image

**Pull request merged**